syntax = "proto3";

package gaiachain;

message Agent {
  // Public key for auth
  string public_key = 1;

  // Email
  string email = 2;

  enum Role {
    SUPER_USER = 0;
    DIRECTOR = 1;
    LOGGER = 2;
    CARBONIZER = 3;
  }
  // Agent role
  Role role = 3;

  // Unix UTC timestamp of approximately when this agent was registered
  uint64 timestamp = 4;

}

message AgentContainer {
  repeated Agent entries = 1;
}
